### PR TITLE
Hw second part spring boot web

### PR DESCRIPTION
**Pull request by pikador10**

### CODE REVIEW

File: src/main/java/com/hw/bookstore/domain/repository/BookRepositoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    @Override
+    public Book findById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("from Book where id = :id", Book.class)
+                    .setParameter("id", id)
+                    .getSingleResult();
+        } catch (NoResultException ex) {
+            throw new BookNotFoundException("Book was not found with id: '%s'"
```
Comment (liliia-ponomarenko) [inline src/main/java/com/hw/bookstore/domain/repository/BookRepositoryImpl.java]: Let's create a general EntityNotFoundException.

File: src/main/java/com/hw/bookstore/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+import java.math.BigDecimal;
+
+public record CreateBookRequestDto(
+        String title,
+        String author,
+        String isbn,
+        BigDecimal price,
+        String description,
+        String coverImage
```
Comment (liliia-ponomarenko) [inline src/main/java/com/hw/bookstore/dto/CreateBookRequestDto.java:11]: here you should validate RequestDto (instead entity validation). Don't forget about annotation Valid in the corresponding controller

File: src/main/java/com/hw/bookstore/service/BookServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    @Override
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toBookDto)
+                .toList();
+    }
+
+    @Override
+    public BookDto findById(Long id) {
+        Book book = bookRepository.findById(id);
```
Comment (liliia-ponomarenko) [inline src/main/java/com/hw/bookstore/service/BookServiceImpl.java:35]: you should return Optional from the repository level and throw EntityNotFoundException here

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution;)
Also, specify the previous branch as a base instead of master, so that changes relevant to this task are displayed in Files changed.

File: src/main/java/com/hw/bookstore/domain/repository/BookRepositoryImpl.java (hunk 1 above)
Comment (pikador10) [reply src/main/java/com/hw/bookstore/domain/repository/BookRepositoryImpl.java]: Done. Renamed to EntityNotFoundException

File: src/main/java/com/hw/bookstore/service/BookServiceImpl.java (hunk 3 above)
Comment (pikador10) [reply src/main/java/com/hw/bookstore/service/BookServiceImpl.java:35]: I catch EntityNotFoundExceptionin in the BookRepositoryImpl class ,  so Optional is redundant in this case.

File: src/main/java/com/hw/bookstore/dto/CreateBookRequestDto.java (hunk 2 above)
Comment (pikador10) [reply src/main/java/com/hw/bookstore/dto/CreateBookRequestDto.java:11]: I'll add validation in the next PRs.

**Review by liliia-ponomarenko — APPROVED** [review]
Great! But let's fix check style issue before merging ;)

**Pull request merged**